src/main/java/AddCommand.java

```diff
@@ -0,0 +1,140 @@
+                return;
+            }
+            break;
+        case "deadline":
+            try {
+                createDeadlineTask(taskList, restOfCommand);
+            } catch (DukeException e) {
+                ui.showInvalidDeadlineError();
+                return;
+            }
+            break;
+        case "event":
+            try {
+                createEventTask(taskList, restOfCommand);
+            } catch (DukeException e) {
+                ui.showInvalidEventError();
+                return;
+            }
+            break;
+        }
+        ui.showTaskIsAdded();
+        ui.showCurrTask(taskList, taskList.numOfTasks);
+        taskList.incrementNumOfTasks();
+        ui.showCurrNumOfTask(taskList);
+    }
+
+    /**
+     * Add a new Todo task to the array of tracked tasks.
+     * @param taskList TaskList object containing the array of tracked tasks
+     * @param input A string containing data regarding the Todo task to be added
+     * @throws DukeException If input has missing information
+     */
+    public static void createTodoTask(TaskList taskList, String input) throws DukeException {
+        if ((input.trim()).length() == 0) {
+            // Task has no description
+            throw new DukeException();
+        }
+        taskList.addTask(new Todo(input));
+    }
+
+    /**
+     * Add a new Deadline task to the array of tracked tasks.
+     * @param taskList TaskList object containing the array of tracked tasks
+     * @param input A string containing data regarding the Deadline task to be added
+     * @throws DukeException If input has missing information
+     */
+    public static void createDeadlineTask(TaskList taskList, String input) throws DukeException {
+        // Extract description and by parts from input
+        int byIndex = input.indexOf("/by");
+        if (byIndex == -1 || byIndex == 0) {
+            // /by does not exist or task does not have description
+            throw new DukeException();
+        }
+        String description = (input.substring(0, byIndex)).trim();
+        String restOfInput = input.substring(byIndex);
+
+        int startingIndex = (restOfInput).indexOf(" ");
+        if (startingIndex == -1) {
+            // /by has no description
+            throw new DukeException();
+        }
+        String by = (restOfInput.substring(startingIndex)).trim();
+        if (by.length() == 0) {
+            // /by's description only has spaces
+            throw new DukeException();
+        }
+        taskList.addTask(new Deadline(description, by));
+    }
+
+    /**
+     * Add a new Event task to the array of tracked tasks.
+     * @param taskList TaskList object containing the array of tracked tasks
+     * @param input A string containing data regarding the Event task to be added
+     * @throws DukeException If input has missing information
+     */
+    public static void createEventTask(TaskList taskList, String input) throws DukeException {
```

Comment:
This function is quite long (> 30 lines). Consider shortening it by refactoring the code inside it to a new method. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)